### DISCUSSION TITLE
Expose API to Prowlarr

### DISCUSSION THREAD

**Discussion by desi76** [autobrr/autobrr#1860] [Ideas]
Currently, I've configured freeleech torrents imported by Autobrr by the category: "free-leech".  This is done, in part, so that freeleech imports do not show as orphaned torrents in the arr clients, but, also, so they can be easily identified in the torrent client.

This works great, except when I attempt to add or search for content in an arr client, that is *already* hosted in the torrent client as "free-leech".  This is because the arr client is only looking for items identified as "tv-sonarr", "movies-radarr", "music-lidarr", etc.  This means, the arr client has to go out and complete a search for and download a new torrent.

Currently, I can get around this by simply adding the series or movie to Sonarr or Radarr, but not searching, then changing the category of the "free-leech" torrent to "tv-sonarr" or "movies-radarr" so that the arr client sees the torrent, parses it and imports it from the torrent client.

Is it possible to expose Autobrr's API to Prowlarr as an indexer, so searches are made to Autobrr as well?  Successful searches would result in Autobrr changing the torrent category of a matching torrent from "free-leech" to "tv-sonarr" or "movies-radarr" allowing the item to be readily imported from the torrent client, avoiding an unnecessary download.

I suppose this would also require Autobrr to actively monitor torrents, which could be useful in other ways.

**Comment by zze0s** [comment]
Okay so lets see,

- You download random releases that is freeleech which are added directly to your client with cat `free-leech`
- You add a new show not previously monitored in any of your arrs
- No auto search because it might redownload what you already have, so you manually look in the client and change category so it can be imported

Is that correct? Sounds like a pretty normal setup and something that is not straightforward to automate.

Implementing torznab feeds for other software to call is a lot of work and I don't think there will be a big benefit in doing so, especially when (pretty much) all indexers autobrr supports are supported in prowlarr/jackett. Your idea would also include deep client integration to fetch torrents and control them. That's an even bigger undertaking.

**Reply by desi76** [reply]
Yes, you summarized my current process correctly.

It all hinges on autobrr actively monitoring imported torrents, so this proposed suggestion may only make sense if autobrr was already considering a change to actively monitor torrents in the client that is already configured in the settings.
